api/role.go

```diff
@@ -84,44 +84,67 @@ func (c CvpRestAPI) GetAllRoles(start, end int) (*RoleList, error) {
 	return &roles, nil
 }
 
-// GetRole returns the role with ID 'roleID'
-func (c CvpRestAPI) GetRole(roleID string) (*SingleRole, error) {
+// GetRoleByID returns the role with ID 'roleID'
+func (c CvpRestAPI) GetRoleByID(roleID string) (*SingleRole, error) {
 	var info SingleRole
 
 	query := &url.Values{"roleId": {roleID}}
 
 	resp, err := c.client.Get("/role/getRole.do", query)
 	if err != nil {
-		return nil, errors.Errorf("GetRole: %s", err)
+		return nil, errors.Errorf("GetRoleByID: %s", err)
 	}
 
 	if err = json.Unmarshal(resp, &info); err != nil {
-		return nil, errors.Errorf("GetRole: %s Payload:\n%s", err, resp)
+		return nil, errors.Errorf("GetRoleByID: %s Payload:\n%s", err, resp)
 	}
 
 	if err := info.Error(); err != nil {
 		// Entity does not exist
 		if info.ErrorCode == "132801" {
 			return nil, nil
 		}
-		return nil, errors.Errorf("GetRole: %s", err)
+		return nil, errors.Errorf("GetRoleByID: %s", err)
 	}
 	return &info, nil
 }
 
+// GetRoleByName returns a role having name 'roleName'
+// by querying for all the roles and returning one that matches
+func (c CvpRestAPI) GetRoleByName(roleName string) (*SingleRole, error) {
+	allRoles, err := c.GetAllRoles(0, 0)
+	if err != nil {
+		return nil, errors.Errorf("GetRoleByName: [%s]", err)
+	}
+	var retRole *SingleRole
+	for _, role := range allRoles.Roles {
+		if role.Name == roleName {
+			retRole = &SingleRole{
+				RoleData: role,
+			}
+			break
```

Comment:
Why not return (retRole, nil) here and remove the if statement below.

Reply:
Yep, that sounds right.